src/modules/core/producer_colour.c

```diff
@@ -187,6 +188,17 @@ static int producer_get_image(mlt_frame frame,
                     *p++ = color.a;
                 }
                 break;
+            case mlt_image_rgba64: {
+                uint16_t *p16 = (uint16_t *) p;
+                const int pixel_count = *width * *height;
+                for (int j = 0; j < pixel_count; j++) {
+                    *p16++ = color.r << 8;
+                    *p16++ = color.g << 8;
+                    *p16++ = color.b << 8;
+                    *p16++ = color.a << 8;
```

Comment:
Let's use array indexing in new code instead of like `*p16++`: `p16[j + 1]` etc.

Reply:
Good idea. Done.